File: pa5/DictionaryTest.c
//Rami Alrwais
//raalrwai
//cmps12b
//pa5
//DictionaryTest.c
#include<stdio.h>
#include<stdlib.h>
#include<string.h>
#include"Dictionary.h"

#define MAX_LEN 180

int main(int argc, char* argv[]){
  Dictionary A = newDictionary();
  char* u;
  char* w;
  char* case1[] = {"this","is","quite", "confusing"};
  char* case2[] = {"one","two","three", "four"};
  int i;
  for(i=0; i<4; i++){
    insert(A, case1[i], case2[i]);
  }
  printDictionary(stdout, A);
  for(i=0; i<4; i++){
    u = case1[i];
    w = lookup(A, k);
    printf("key=\"%s\" %s\"%s\"\n", u, (w==NULL?"not found ":"value="), w);
  }
  delete(A, "one");
  delete(A, "three");
  printDictionary(stdout, A);
  for(i=0; i<4; i++){
    u = case1[i];
    w = lookup(A, u);
    printf("key=\"%s\" %s\"%s\"\n", u, (w==NULL?"not found ":"value="), w);
    printf("%s\n", (isEmpty(A)?"true":"false"));
    printf("%d\n", size(A));
    makeEmpty(A);
    printf("%s\n", (isEmpty(A)?"true":"false"));
    freeDictionary(&A);
    return(EXIT_SUCCESS);
  }
}

File: pa3/DictionaryTest.java
//Rami Alrwais
//raalrwai
//cmps12b
//DictionaryTest
//Tests out all the functions of Dictionary.java in 
public class DictionaryTest {

  public static void main(String[] args) {
    Dictionary A = new Dictionary();
    System.out.println(A.isEmpty());
    A.insert("1","f");
    System.out.println(A.isEmpty());
    A.insert("5","h");
    System.out.println(A.size());
    A.insert("3","a");
    System.out.println("*********");
    System.out.println(A.size());
    System.out.println("*********");
    A.insert("8","c");
    A.insert("2","e");
    A.insert("9","3");
    //A.delete("4");
    //A.insert("7","e");
    //throws exception
    A.delete("1");
    System.out.println(A);
    System.out.println(A.size());
    A.delete("3");
    System.out.println(A.size());
    System.out.println(A);
    System.out.println(A.lookup("5"));
    A.insert("1", "f"); //throws DuplicateKeyException
   // A.delete("10"); //throws KeyNotFoundException
   // A.makeEmpty();
    System.out.println(A.size() + " " + A.isEmpty());
  }
}
